suite-native/module-home/src/screens/HomeScreen/components/RememberDeviceBottomSheet.tsx

```diff
@@ -0,0 +1,136 @@
+import { BottomSheet, Box, Button, Text, VStack } from '@suite-native/atoms';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { useIsBiometricsInitialSetupFinished } from '@suite-native/biometrics';
+import { Translation } from '@suite-native/intl';
+import {
+    selectRememberModeOfferChoiceTimestamp,
+    setRememberModeOfferChoiceTimestamp,
+} from '@suite-native/module-settings';
+import { useActiveColorScheme } from '@suite-native/theme';
+
+import { DisconnectedTrezorDarkSvg } from '../../../assets/DisconnectedTrezorDarkSvg';
+import { DisconnectedTrezorSvg } from '../../../assets/DisconnectedTrezorSvg';
+
+const SHOW_TIMEOUT = 1500;
+
+const svgStyle = prepareNativeStyle(utils => ({
+    paddingTop: utils.spacings.small,
+    width: '100%',
+    height: 206,
+}));
+const textContentStyle = prepareNativeStyle(utils => ({
+    gap: utils.spacings.small,
+}));
+
+const buttonWrapperStyle = prepareNativeStyle(utils => ({
+    gap: utils.spacings.medium,
+}));
+
+export const RememberDeviceBottomSheet = () => {
+    const dispatch = useDispatch();
+    const { applyStyle } = useNativeStyles();
+    const colorScheme = useActiveColorScheme();
+    const { isBiometricsInitialSetupFinished } = useIsBiometricsInitialSetupFinished();
+    const isDeviceAuthorized = useSelector(selectIsDeviceAuthorized);
+    const device = useSelector(selectDevice);
+    const isPortfolioTrackerDevice = useSelector(selectIsPortfolioTrackerDevice);
+    const rememberModeOfferChoiceTimestamp = useSelector(selectRememberModeOfferChoiceTimestamp);
+    const isDeviceInViewOnlyMode = useSelector(selectIsDeviceInViewOnlyMode);
+
+    const [isVisible, setIsVisible] = useState(false);
+
+    // show the bottom sheet if:
+    //     does not have view only mode enabled already
+    //     the device is authorized
+    //     not a portfolio tracker
+    //     the user hasn't made a choice yet
+    //     and biometrics initial setup was decided
+
+    const canBeShowed =
+        !isDeviceInViewOnlyMode &&
+        isDeviceAuthorized &&
+        !isPortfolioTrackerDevice &&
+        !rememberModeOfferChoiceTimestamp &&
+        isBiometricsInitialSetupFinished;
+
+    useEffect(() => {
+        let isMounted = true;
+        let timerId: ReturnType<typeof setTimeout>;
+
+        //show after a delay
+        if (canBeShowed) {
+            timerId = setTimeout(() => {
+                if (isMounted) {
+                    setIsVisible(true);
+                }
+            }, SHOW_TIMEOUT);
+        }
+
+        return () => {
+            clearTimeout(timerId);
+            isMounted = false;
+        };
+    }, [canBeShowed]);
+
+    const handleSetRememberModeOfferChoiceTimestamp = () =>
+        dispatch(setRememberModeOfferChoiceTimestamp());
+
+    const handleClose = () => {
+        handleSetRememberModeOfferChoiceTimestamp();
+    };
+
+    const handleEnable = () => {
+        handleSetRememberModeOfferChoiceTimestamp();
```

Comment:
I would not expect setting timestamp on enable and probably not even on close (only on clicking on skip) and I believe it is described like that in the issue description.

Reply:
73ad19a3b8